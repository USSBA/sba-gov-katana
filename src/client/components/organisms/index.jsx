/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
