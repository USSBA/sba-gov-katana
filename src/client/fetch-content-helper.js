import axios from 'axios'
import path from 'path'
import queryString from 'querystring'

async function fetchRestContent(id, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content' + (id ? '/' + id : '') + '.json',
      langOverride && { headers: { 'accept-language': langOverride } }
    )

    if (langOverride === 'es') {
      data = response.data.spanishTranslation
    } else {
      data = response.data
    }
  } catch (error) {
    console.error('fetchRestContent', error)
  }

  return data
}

async function fetchApiDistrictOfficeName(zipcode) {
  let data = null
  try {
    const response = await axios.get(
      `https://h4z2sm4z28.execute-api.us-east-1.amazonaws.com/default/dev-ryan-district-office-zipcode-api?zipcode=${zipcode}`
    )
    data = response.data
  } catch (error) {
    console.error('fetchDistrictOfficeName', error)
  }
  console.log('the data is', data)
  return data
}

async function fetchApiSbaGovContent(fileDirectory) {
  let data = null

  try {
    const response = await axios.get(`https://api.sba.gov/${fileDirectory}`)

    data = JSON.parse(response.data)
  } catch (error) {
    console.error('fetchApiSbaGovContent', error)
  }

  return data
}

async function fetchEventContent(id, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content/search/event' + (id ? '/' + id : '') + '.json',
      langOverride && { headers: { 'accept-language': langOverride } }
    )
    data = response.data
  } catch (error) {
    console.error('fetchEventContent', error)
  }

  return data
}

async function fetchSiteContent(type, query, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content/search/' + type + '.json' + (query ? '?' + queryString.stringify(query) : ''),
      langOverride && { headers: { 'accept-language': langOverride } }
    )
    data = response.data
  } catch (error) {
    console.error('fetchSiteContent', error)
  }
  return data
}

async function postMiscAction(endpoint, body) {
  const url = path.join('/actions/misc', `${endpoint}`)

  let data = null

  try {
    const response = await axios.post(url, body)
    data = response.data
  } catch (error) {
    console.error('postMiscAction', error)
    throw error
  }

  return data
}

async function runMiscAction(type, query) {
  const url = '/actions/misc/' + type + '.json' + (query ? '?' + queryString.stringify(query) : '')
  let data = null
  try {
    const response = await axios.get(url)
    data = response.data
  } catch (error) {
    console.error('fetchSiteContent', error)
  }
  return data
}

export {
  fetchApiDistrictOfficeName,
  fetchApiSbaGovContent,
  fetchRestContent,
  fetchEventContent,
  fetchSiteContent,
  postMiscAction,
  runMiscAction
}
