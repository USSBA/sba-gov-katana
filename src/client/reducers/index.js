import {combineReducers} from 'redux';
import {routerReducer} from 'react-router-redux';

// our customer reducers
import loanReducer from '../reducers/loan-form.js';
import additionalInfoReducer from '../reducers/additional-info.js';
import contactInfoReducer from '../reducers/contact-info.js';
import businessInfoReducer from '../reducers/business-info.js';
import reviewSubmitInfoReducer from './review-submit-info.js';
import industryInfoReducer from './industry-info.js';

// combine the custom reducers with the routing reducer
const rootReducer = combineReducers({
  loanReducer: loanReducer,
  additionalInfoReducer: additionalInfoReducer,
  contactInfoReducer: contactInfoReducer,
  businessInfoReducer: businessInfoReducer,
  reviewSubmitInfoReducer: reviewSubmitInfoReducer,
  industryInfoReducer: industryInfoReducer,
  routing: routerReducer
});

export default rootReducer;
