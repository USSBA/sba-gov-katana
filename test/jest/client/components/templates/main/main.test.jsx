import React from 'react'
import { shallow } from 'enzyme'

import Main from 'client/components/templates/main'
import * as helper from 'client/fetch-content-helper'
import { Header } from 'organisms'

describe('Main', () => {
  const disasterAlert = {
    buttonText: 'Learn more',
    description: 'Get information about disaster assistance',
    link: '/disaster-assistance',
    visible: false
  }

  test('should render the happy path', () => {
    const tree = shallow(<Main />)
    tree.setState({ disasterAlert })
    expect(tree).toMatchSnapshot()
  })

  test('should render with a header', () => {
    const tree = shallow(<Main />)
    tree.setState({ disasterAlert })
    expect(tree.find(Header).length).toEqual(1)
  })
})
