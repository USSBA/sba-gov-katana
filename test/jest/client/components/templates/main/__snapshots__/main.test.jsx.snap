// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render the happy path 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={false}
>
    <DisasterAlert
        buttonText="Learn more"
        description="Get information about disaster assistance"
        link="/disaster-assistance"
        onClose={[Function]}
        spanishTranslation={undefined}
        visible={false}
    />
    <a
        className="skipNav"
        href="#main-content"
    >
        Skip to main content
    </a>
    <Connect(Header)
        additionalMenuOffset={0}
    />
    <Connect(MainLoader) />
    <div
        className="mainContent"
        id="main-content"
    />
    <Footer />
    <Connect(ModalController) />
</div>,
  "nodes": Array [
    <div
      className={false}
>
      <DisasterAlert
            buttonText="Learn more"
            description="Get information about disaster assistance"
            link="/disaster-assistance"
            onClose={[Function]}
            spanishTranslation={undefined}
            visible={false}
      />
      <a
            className="skipNav"
            href="#main-content"
      >
            Skip to main content
      </a>
      <Connect(Header)
            additionalMenuOffset={0}
      />
      <Connect(MainLoader) />
      <div
            className="mainContent"
            id="main-content"
      />
      <Footer />
      <Connect(ModalController) />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Main />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Main {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "disasterAlert": Object {
            "buttonText": "Learn more",
            "description": "Get information about disaster assistance",
            "link": "/disaster-assistance",
            "visible": false,
          },
          "disasterAlertCookieExists": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={false}
>
          <DisasterAlert
                    buttonText="Learn more"
                    description="Get information about disaster assistance"
                    link="/disaster-assistance"
                    onClose={[Function]}
                    spanishTranslation={undefined}
                    visible={false}
          />
          <a
                    className="skipNav"
                    href="#main-content"
          >
                    Skip to main content
          </a>
          <Connect(Header)
                    additionalMenuOffset={0}
          />
          <Connect(MainLoader) />
          <div
                    className="mainContent"
                    id="main-content"
          />
          <Footer />
          <Connect(ModalController) />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={false}
>
          <DisasterAlert
                    buttonText="Learn more"
                    description="Get information about disaster assistance"
                    link="/disaster-assistance"
                    onClose={[Function]}
                    spanishTranslation={undefined}
                    visible={false}
          />
          <a
                    className="skipNav"
                    href="#main-content"
          >
                    Skip to main content
          </a>
          <Connect(Header)
                    additionalMenuOffset={0}
          />
          <Connect(MainLoader) />
          <div
                    className="mainContent"
                    id="main-content"
          />
          <Footer />
          <Connect(ModalController) />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Main />,
}
`;
